# HG changeset patch
# User Rik <rik@octave.org>
# Date 1515193023 28800
#      Fri Jan 05 14:57:03 2018 -0800
# Node ID ed9a040873282543de1859e4e2aa3cf387b57e77
# Parent  e1295dd49c71b89d2dbaed675b7ba5fafc7dd99e
test: Use new constant for acos(2) in BIST tests of acos.

* mappers.cc: Use Mathematica variable precision to create an 18-digit constant
for acos(2) used in BIST tests.

diff --git a/libinterp/corefcn/mappers.cc b/libinterp/corefcn/mappers.cc
--- a/libinterp/corefcn/mappers.cc
+++ b/libinterp/corefcn/mappers.cc
@@ -108,17 +108,17 @@ Compute the inverse cosine in radians fo
 %!test
 %! x = single ([1, rt3/2, rt2/2, 1/2, 0, -1/2, -rt2/2, -rt3/2, -1]);
 %! v = single ([0, pi/6, pi/4, pi/3, pi/2, 2*pi/3, 3*pi/4, 5*pi/6, pi]);
 %! assert (acos (x), v, sqrt (eps ("single")));
 
 ## Test values on either side of branch cut
 %!test
 %! rval = 0;
-%! ival = 1.31695789692481635;
+%! ival = 1.31695789692481671;
 %! obs = acos ([2, 2-i*eps, 2+i*eps]);
 %! exp = [rval + ival*i, rval + ival*i, rval - ival*i];
 %! assert (obs, exp, 2*eps);
 %! rval = pi;
 %! obs = acos ([-2, -2-i*eps, -2+i*eps]);
 %! exp = [rval - ival*i, rval + ival*i, rval - ival*i];
 %! assert (obs, exp, 2*eps);
 %! assert (acos ([2 0]),  [ival*i, pi/2], 2*eps);
